### PR TITLE
chore: Move `kedro-catalog` JSON schema to `kedro-datasets`

### PR DESCRIPTION
**Pull request by chrisschopp**
## Description
Resolving [4258](https://github.com/kedro-org/kedro/issues/4258)

## Development notes
Moved `static/jsonschema/` from `kedro` (there is a [corresponding PR there](https://github.com/kedro-org/kedro/pull/4359)) to `kedro-datasets`.

## Checklist

- [X] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [X] Updated the documentation to reflect the code changes
- [X] Added a description of this change in the relevant `RELEASE.md` file
- [X] Added tests to cover my changes
- [ ] Received approvals from at least half of the TSC (required for adding a new, non-experimental dataset)

### CODE REVIEW

File: kedro-datasets/static/jsonschema/kedro-catalog-0.19.json [hunk 1]
```diff
@@ -0,0 +1,1470 @@
...
+    "patternProperties": {
+      "^[a-z0-9-_]+$": {
+        "required": [
+          "type"
+        ],
+        "properties": {
+          "type": {
+            "type": "string",
+            "enum": [
+              "CachedDataset",
```
Comment (merelcht) [inline kedro-datasets/static/jsonschema/kedro-catalog-0.19.json:12]: CachedDataset, MemoryDataset and LambdaDataset are still part of the Kedro core package, so I'm not sure they should actually be part of this list..

File: kedro-datasets/static/jsonschema/kedro-catalog-0.19.json [hunk 2]
```diff
@@ -0,0 +1,1470 @@
...
+              "IncrementalDataset",
```
Comment (merelcht) [inline kedro-datasets/static/jsonschema/kedro-catalog-0.19.json:13]:
```suggestion
              "partitions.IncrementalDataset",
```

File: kedro-datasets/static/jsonschema/kedro-catalog-0.19.json [hunk 3]
```diff
@@ -0,0 +1,1470 @@
...
+                  "description": "Extra arguments to pass into underlying filesystem class constructor\n(e.g. `{\"project\": \"my-project\"}` for ``GCSFileSystem``)"
+                }
+              }
+            }
+          },
+          {
+            "if": {
+              "properties": {
+                "type": {
+                  "const": "MemoryDataset"
```
Comment (merelcht) [inline kedro-datasets/static/jsonschema/kedro-catalog-0.19.json:134]: More references here to datasets that aren't part of `kedro-datasets`.

**Review by merelcht — COMMENTED** [review]
Thanks for the contribution @chrisschopp! This mostly looks good, but now some of the datasets are still part of the core Kedro package, so I don't think those should now still be in the list here.

**Review by merelcht — APPROVED** [review]
Thanks @chrisschopp ! LGTM 👍 

Can you add the change to the release notes as well? Together with your name in the list of contributors: https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/RELEASE.md#community-contributions

**Review by ankatiyar — APPROVED** [review]
Thanks @chrisschopp!

**Pull request merged**